controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -273,16 +274,25 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.scaleDownControlPlane(ctx, cluster, kcp, ownedMachines)
 	}
 
-	// Update kube-proxy image name
+	// Get the workload cluster client.
 	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
 	if err != nil {
-		// Just log and continue, control-plane cluster may be uninitialized
+		// Just log and continue, control-plane cluster may be uninitialized.
 		logger.Error(err, "failed to get remote client for workload cluster")
```

Comment:
Now that we are doing this after the switch, it might be good to go ahead and return the error here now (that or return `ctrl.Result{Requeue: true}`). Currently we are essentially doing this, since the deferred call to `updateStatus` is doing this for us when it attempts to call `GetWorkloadCluster` itself, but I think it might be explictly do it here rather than rely on `updateStatus` to ensure that we requeue additional times until we can complete these steps.

Reply:
What do you think about removing the log line or doing making it info if we requeue?

Reply:
+1 to info or removing the log line